src/render/tutorials/caching-http-requests/2_0.html.md

```diff
@@ -0,0 +1,319 @@
+- `ports` - port accessible from `localhost` is specified on the left side, and a port available in the virtual environment is specified on the right 
+- `networks` - specifies in which docker network the service will be deployed.
+
+Then we have `knotx` service specification. It's very similar to the previous one, so only one property is worth explaining. 
+- `command` - this property is used to specify Knot.x starting commands.
+
+Next step in our configuration is to set `rootProject.name` in `settings.gradle.kts` (also at the project level directory),
+also make sure that `healh-check` is included 
+```gradle
+rootProject.name = "api-cache"
+
+include("health-check")
+
+project(":health-check").projectDir = file("modules/health-check")
+``` 
+
+Next thing to do is to configure `gradle.properties` at project level directory. Let's fill it with
+```gradle
+version=2.0.0-SNAPSHOT
+knotx.version=2.0.0
+knotx.conf=knotx
+docker.image.name=knotx-example/api-cache
+```
+Very important step here is to set `docker.image.name`, which is necessary to run proper docker image.
+
+`build.gradle.kts` file could stay as it was provided with starterkit, we don't have to apply any modifications here.
+The same if it comes to `./docker/Dockerfile`, version provided by default is absolutely enough for our purposes.
+
+## Wiremock 
+Wiremock is simply used for mocking server, we don't have to create real REST API, for our training purposes it's enough.
+To do it let's create following directories on the project level directory:
+- `common-services/webapi/__files/`
+- `common-services/webapi/mappings/`
+
+Into first directory please paste `product.json` file with following body:
+```json
+{
+  "id": 21762532,
+  "url": "http://knotx.io",
+  "label": "Product"
+}
+```
+
+And into second directory please paste `product.json` file with following body:
+```json
+{
+  "request": {
+    "method": "GET",
+    "url": "/product/id"
+  },
+  "response": {
+    "status": 200,
+    "fixedDelayMilliseconds": 100,
+    "bodyFileName": "product.json"
+  }
+}
+```
+
+Make sure that in `api-cache.yml` property `volume` properly points to `webapi` directory. And that's all about Wiremock.
+
+## Build and run
+From project level directory execute following commands which will build project and run Docker Swarm:
+```shell script
+$ gradlew clean build
+$ docker swarm init
+$ docker stack deploy -c api-cache.yml api-cache
+```
+After successful execution you should be able to see http response at [localhost:8092/product/id](localhost:8092/product/id)
+
+## Summary of what we actually did
+To sum it up in this tutorial we've configured in-memory cache and show how easily we can replace it with another one.
+What's important we did it without writing any line of code, but only using Knot.x configs. We've focused on creating it
+in declarative way instead of implementing cache on our own. Hopefully, now you'll know how easily you can extend other 
```

Comment:
`Hopefully ... ` sounds quite bad, it suggests that the reader might not understand the tutorial

Reply:
fixed